cardano-api/src/Cardano/Api/IPC/NtcVersionOf.hs

```diff
@@ -0,0 +1,10 @@
+module Cardano.Api.IPC.NtcVersionOf
+  ( NtcVersionOf (..)
+  ) where
+
+import           Ouroboros.Network.NodeToClient.Version (NodeToClientVersion (..))
+
+-- | The query 'a' is a versioned query, which means it requires the Node to support a minimum
+-- Node-to-Client version.
```

Comment:
We probably don't need a type class for this. We can implement this as a function in the `Query` module.

Reply:
If we implement this as a function, it won't be one function, but a function per type.  Something like `ntcVersionOfQueryTypeName` where `QueryTypeName` is the name of the type.  Is that okay?

Reply:
Converting from function to typeclass also gives users the opportunity to not use our function but substitute their own, which most likely will be wrong too.

Reply:
> but a function per type

I've had a second look and I think you're right :+1: